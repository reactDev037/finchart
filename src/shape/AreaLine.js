import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Line from './Line';
import Area from './Area';

export class Component extends PureComponent {
  static propTypes = {
    className: PropTypes.string.isRequired,
    xScale: PropTypes.func.isRequired,
    yScale: PropTypes.func.isRequired,
    height: PropTypes.number.isRequired,
    width: PropTypes.number.isRequired,
    data: PropTypes.arrayOf(PropTypes.object).isRequired,
    fields: PropTypes.shape({
      x: PropTypes.string.isRequired,
      y: PropTypes.string.isRequired,
    }).isRequired,
    // defined: PropTypes.array,
    // curve: PropTypes.string,
  };

  static defaultProps = {
    curve: 'curveLinear',
  };

  render() {
    const {
      className,
      xScale,
      yScale,
      height,
      width,
      data,
      fields,
      // defined,
      // curve,
    } = this.props;

    return (
      <g className={`arealine-${className}`}>
        <Line {...{ height, width, xScale, yScale, data, fields }} />
        <Area {...{ height, width, xScale, yScale, data, fields }} />
      </g>
    );
  }
}

export default styled(Component)`
`;
