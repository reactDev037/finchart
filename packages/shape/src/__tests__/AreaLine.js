import React from 'react';
import { scaleLinear } from 'd3-scale';
import { render } from 'enzyme';
import AreaLine from '../AreaLine';

describe('<AreaLine />', () => {
  it('handles data input format error', () => {
    const width = 960;
    const height = 360;
    const xScale = scaleLinear().domain([0, 100]).range([0, width]);
    const yScale = scaleLinear().domain([0, 100]).range([height, 0]);
    const data = [
      { supply: 0, price: 100 },
      { supply: 30, price: 90 },
      { supply: 100, price: 0 },
    ];
    const fields = {
      x: 'volume',
      y: 'price',
    };
    const wrapper = render(
      <AreaLine {...{ height, width, xScale, yScale, data, fields }} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders area', () => {
    const width = 960;
    const height = 360;
    const xScale = scaleLinear().domain([0, 100]).range([0, width]);
    const yScale = scaleLinear().domain([0, 100]).range([height, 0]);
    const data = [
      { supply: 0, price: 100 },
      { supply: 30, price: 90 },
      { supply: 100, price: 0 },
    ];
    const fields = {
      x: 'supply',
      y: 'price',
    };
    const wrapper = render(
      <AreaLine {...{ height, width, xScale, yScale, data, fields }} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('handles missing data properly');

  it('handles partial data key pollution');
});
