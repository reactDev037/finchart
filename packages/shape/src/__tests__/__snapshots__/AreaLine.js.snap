// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AreaLine /> handles data input format error 1`] = `
<g
  class="arealine-sc-htpNat hHtpcc"
>
  <text>
    Input Data Format Error
  </text>
  <text>
    Input Data Format Error
  </text>
</g>
`;

exports[`<AreaLine /> renders area 1`] = `
<g
  class="arealine-sc-htpNat hHtpcc"
>
  <path
    class="line-sc-bdVaJa gxPDDw"
    d="M0,0L288,36L960,360"
  />
  <path
    class="area-sc-bwzfXH lbUVb"
    d="M0,0L288,36L960,360L960,360L288,360L0,360Z"
  />
</g>
`;
